File: backend-frontend/bff-client/src/entities/Loan.ts
import { CurrencyType } from 'entities/Currency'
import { Tariff } from 'entities/Tariff'

export type Loan = {
  id: string
  userId: string
  accountId: string
  tariff: Tariff
  lastChargeDate?: string | null
  currencyType: CurrencyType
  debt: number
  isClosed: boolean
}

File: backend/Loan/Loan.DAL/Entities/Loan.cs
using System.ComponentModel.DataAnnotations;
using Common.Enum;
using Common.Persistence;

namespace Loan.DAL.Entities;

public class Loan : BaseEntity
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid UserId { get; set; }

    public Guid AccountId { get; set; }

    public Tariff Tariff { get; set; }
    public List<PaymentRequest> Payments { get; set; } = [];
    public DateTime? LastChargeDate { get; set; }

    public CurrencyType CurrencyType { get; set; }
    public bool IsClosed { get; set; } = false;

    public long Debt { get; set; }
}
